services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/HankeServiceITests.kt

```diff
@@ -1201,6 +1204,90 @@ class HankeServiceITests(
         }
     }
 
+    @Nested
+    inner class GenerateHankeWithJohtoselvityshakemus {
+        private val hakemusNimi = "Johtoselvitys for a private property"
+
+        @Test
+        fun `saves hanke based on request`() {
+            val request = CreateHankeRequest(hakemusNimi, DEFAULT_HANKE_PERUSTAJA)
+
+            hankeService.generateHankeWithJohtoselvityshakemus(request, setUpProfiiliMocks())
+
+            assertThat(hankeRepository.findAll()).single().all {
+                prop(HankeEntity::generated).isTrue()
+                prop(HankeEntity::status).isEqualTo(HankeStatus.DRAFT)
+                prop(HankeEntity::nimi).isEqualTo(hakemusNimi)
+                prop(HankeEntity::createdByUserId).isEqualTo(USERNAME)
+                prop(HankeEntity::createdAt).isRecentUTC()
+            }
+        }
+
+        @Test
+        fun `saves hakemus based on request`() {
+            val request = CreateHankeRequest(hakemusNimi, DEFAULT_HANKE_PERUSTAJA)
+
+            hankeService.generateHankeWithJohtoselvityshakemus(request, setUpProfiiliMocks())
+
+            assertThat(applicationRepository.findAll())
+                .single()
+                .prop(ApplicationEntity::applicationData)
+                .prop(ApplicationData::name)
+                .isEqualTo(hakemusNimi)
+        }
+
+        @Test
+        fun `returns the saved hakemus`() {
+            val request = CreateHankeRequest(hakemusNimi, DEFAULT_HANKE_PERUSTAJA)
+
+            val hakemus =
+                hankeService.generateHankeWithJohtoselvityshakemus(request, setUpProfiiliMocks())
+
+            assertThat(hakemus).all {
+                prop(Hakemus::applicationData).prop(HakemusData::name).isEqualTo(hakemusNimi)
+            }
+        }
+
+        @Test
+        fun `generates hankekayttaja for founder`() {
+            val request = CreateHankeRequest(hakemusNimi, DEFAULT_HANKE_PERUSTAJA)
+
+            val hakemus =
+                hankeService.generateHankeWithJohtoselvityshakemus(request, setUpProfiiliMocks())
+
+            val hanke = hankeRepository.findByHankeTunnus(hakemus.hankeTunnus)!!
+            val kayttajat = hankekayttajaRepository.findByHankeId(hanke.id)
+            assertThat(kayttajat.first()).all {
+                prop(HankekayttajaEntity::id).isNotNull()
+                prop(HankekayttajaEntity::sahkoposti).isEqualTo(DEFAULT_HANKE_PERUSTAJA.sahkoposti)
+                prop(HankekayttajaEntity::puhelin).isEqualTo(DEFAULT_HANKE_PERUSTAJA.puhelinnumero)
+                prop(HankekayttajaEntity::etunimi).isEqualTo(ProfiiliFactory.DEFAULT_GIVEN_NAME)
+                prop(HankekayttajaEntity::sukunimi).isEqualTo(ProfiiliFactory.DEFAULT_LAST_NAME)
```

Comment:
Maybe these constants from `ProfiiliFactory` could be in import statements to make it a bit readable here?

Reply:
I imported the constants 'staticly'.